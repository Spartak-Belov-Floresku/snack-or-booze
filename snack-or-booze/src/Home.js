import { Link } from "react-router-dom";
import { 
  Card, 
  CardBody, 
  CardTitle, 
  CardColumns,
  CardText,
  ListGroup,
  ListGroupItem
 } from "reactstrap";


/**
 * component will upload data available drinks and snacks on the home page of the UI
 */
function Home({snacks, drinks}) {
  return (
    <section className="col-md-8">
      <Card>
        <CardBody className="text-center">
          <CardTitle>
            <span className="font-weight-bold">
              Welcome to Silicon Valley's premier dive cafe!
            </span>
          </CardTitle>
        </CardBody>
        <CardBody>
          <CardColumns className="float-left">
            <CardText className="menu-title">Food Menu:</CardText>
            <ListGroup className="home-page-menu">
              {snacks.map(snack => (
                <Link to={`/snacks/${snack.id}`} key={snack.id}>
                  <ListGroupItem>{snack.name}</ListGroupItem>
                </Link>
              ))}
            </ListGroup>
          </CardColumns>
          <CardColumns className="float-right">
            <CardText className="menu-title">Drinks Menu:</CardText>
            <ListGroup className="home-page-menu">
              {drinks.map(drink => (
                <Link to={`/drinks/${drink.id}`} key={drink.id}>
                  <ListGroupItem>{drink.name}</ListGroupItem>
                </Link>
              ))}
            </ListGroup>
          </CardColumns>
        </CardBody>
      </Card>
    </section>
  );
}

export default Home;
