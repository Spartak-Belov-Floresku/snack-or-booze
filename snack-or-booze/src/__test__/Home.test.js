import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import Home from '../Home';
import SnackOrBoozeApi from "../Api";

let snacks, drinks;

/**befor ech tests get data from database */
beforeEach( async () => {
  function getSnacks() {
    return SnackOrBoozeApi.getSnacks();
  }

  function getDrinks(){
    return SnackOrBoozeApi.getDrinks();
  }    
  
  snacks = await getSnacks();
  drinks = await getDrinks();

});


/** render Home component */
test('renders learn react link', () => {

    render(
        <BrowserRouter>
            <Home snacks={snacks} drinks={drinks} />
        </BrowserRouter>
    );
  
});

/** Matching snapshot of Home components */
it("matches snapshot Home component", () => {  
      
    const {asFragment} = render(
        <BrowserRouter>
            <Home snacks={snacks} drinks={drinks} />
        </BrowserRouter>
    );
    
    expect(asFragment()).toMatchSnapshot();
});


/** check if the menu exists on the home page */
it("menu on the home page", () => {

    const { queryByText } = render(
        <BrowserRouter>
            <Home snacks={snacks} drinks={drinks} />
        </BrowserRouter>
    );
    expect(queryByText("Food Menu:")).toBeInTheDocument();
    expect(queryByText("Drinks Menu:")).toBeInTheDocument();

});

/** checking if links to snack and drink exist inmenu bar */
it("check link", () => {

    const { getByText } = render(
        <BrowserRouter>
            <Home snacks={snacks} drinks={drinks} />
        </BrowserRouter>
    );

    expect(screen.getByText('Nachos').closest('a')).toHaveAttribute('href', '/snacks/nachos');
    expect(screen.getByText('Gin and Tonic').closest('a')).toHaveAttribute('href', '/drinks/gin-and-tonic');


});