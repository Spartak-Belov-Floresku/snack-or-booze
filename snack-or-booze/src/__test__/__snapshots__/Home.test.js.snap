// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot Home component 1`] = `
<DocumentFragment>
  <section
    class="col-md-8"
  >
    <div
      class="card"
    >
      <div
        class="text-center card-body"
      >
        <h5
          class="card-title"
        >
          <span
            class="font-weight-bold"
          >
            Welcome to Silicon Valley's premier dive cafe!
          </span>
        </h5>
      </div>
      <div
        class="card-body"
      >
        <div
          class="float-left card-columns"
        >
          <p
            class="menu-title card-text"
          >
            Food Menu:
          </p>
          <ul
            class="home-page-menu list-group"
          >
            <a
              href="/snacks/nachos"
            >
              <li
                class="list-group-item"
              >
                Nachos
              </li>
            </a>
            <a
              href="/snacks/hummus"
            >
              <li
                class="list-group-item"
              >
                Hummus
              </li>
            </a>
            <a
              href="/snacks/arugula-and-walnut-salad"
            >
              <li
                class="list-group-item"
              >
                Arugula and Walnut Salad
              </li>
            </a>
          </ul>
        </div>
        <div
          class="float-right card-columns"
        >
          <p
            class="menu-title card-text"
          >
            Drinks Menu:
          </p>
          <ul
            class="home-page-menu list-group"
          >
            <a
              href="/drinks/martini"
            >
              <li
                class="list-group-item"
              >
                Martini
              </li>
            </a>
            <a
              href="/drinks/negroni"
            >
              <li
                class="list-group-item"
              >
                Negroni
              </li>
            </a>
            <a
              href="/drinks/gin-and-tonic"
            >
              <li
                class="list-group-item"
              >
                Gin and Tonic
              </li>
            </a>
            <a
              href="/drinks/diet-coke"
            >
              <li
                class="list-group-item"
              >
                Diet Coke
              </li>
            </a>
          </ul>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
